### PR TITLE
Hub - check trailing and double slashes when constructing API urls, better page/sort param support

### PR DESCRIPTION
**Pull request by himdel**
Fixes AAH-3007, follows #1408 

`hubAPI` and `pulpAPI` will now throw when any of these happen:
* `HUB_API_PREFIX` ends with a trailing slash
* prefix-relative url doesn't start with a slash (already happening)
* url doesn't end in a slash (end of string or before a ?)
* url contains a double slash
* url contains a double ?
* the wrong param is used based on the url (`page, limit, offset, page_size, sort, ordering, order_by`)

add `hubQueryString` to wrap `normalizeQueryString` and:
* escape values
* convert page/limit/offset/page_size/sort/ordering/sort_by to the right name

allow a `` hubAPI`/foo/?${{ param1: 'abc', param2: 123 }}` `` syntax, calling `hubQueryString`

fixes all instances of `hubAPI` & `pulpAPI` missing a trailing slash

also, this allows us to fix all those `|| ''` and `?? ''` and `as string` in urls

and fix tests

---

In e2e tests, `hubAPI` and `pulpAPI` retain their simpler versions, I'm not sure how to handle the use of `*` wildcards without allowing them in regular code.

### CODE REVIEW

File: cypress/support/formatApiPathForHub.ts [hunk 1]
```diff
@@ -9,7 +9,11 @@
 import { apiTag } from '../../frontend/hub/common/api/formatPath';
 
 function getBaseAPIPath() {
-  return (Cypress.env('HUB_API_PREFIX') as string) || '/api/galaxy';
+  const base = (Cypress.env('HUB_API_PREFIX') as string) || '/api/galaxy';
+  if (base.endsWith('/')) {
+    throw new Error(`Invalid HUB_API_PREFIX - must NOT end with a slash`);
```
Comment (nixocio) [inline cypress/support/formatApiPathForHub.ts:14]: 🥇

File: cypress/e2e/hub/approvals.cy.ts [hunk 2]
```diff
@@ -12,6 +12,7 @@ describe('Approvals', () => {
     cy.hubLogin();
```
Comment (nixocio) [inline cypress/e2e/hub/approvals.cy.ts:12]: @acruzgon Please review the approval tests changes.

**Review by acruzgon — APPROVED** [review]
lgtm 
image

File: cypress/e2e/hub/approvals.cy.ts (hunk 2 above)
Comment (acruzgon) [reply cypress/e2e/hub/approvals.cy.ts:12]: all tests passed
image

**Comment by himdel** [comment]
rebased, and updated the offset <-> page conversion to fix an off-by-one (page starts at 1, offset at 0: https://github.com/ansible/ansible-ui/pull/1530/commits/b600638eab405d8ab74731d06896ec5ad967ff7b)

**Review by ZitaNemeckova — APPROVED** [review]

**Pull request merged**